src/components/slideshow/index.js

```diff
@@ -0,0 +1,340 @@
+      height: 429, // px
+      paddingRight: 4, // px
+    },
+    offset: {
+      left: 30, // px
+    },
+  },
+}
+
+// Assuming there are three images [ A, B, C ] for slideshow.
+// If image B is rendered in the center,
+// users can see part of image A(left side) and image C(right side) with masks.
+// When users click right button to see image C, which means, C is in the center,
+// users still can see part of image B(left side) and image A(right side) with masks.
+//
+// Hence, there are four images rendered arround B at the beginning.
+// The image array should be [ C, A, B, C, A ].
+//
+// `slidesOffset` indicates how many slides rendered before/after image B, which is, 2 (A and C).
+//
+const slidesOffset = 2
+
+// duration of transition of transform(translateX(?px))
+const duration = 300
+
+// current index to indicate which image should be rendered in the center
+const defaultCurIndex = 0
+
+const SlideshowFlexBox = styled.div`
+  max-width: 759px;
+  width: 100%;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  flex-wrap: wrap;
+`
+
+const SlidesSection = styled.div`
+  flex-shrink: 0;
+  flex-basis: 100%;
+  overflow: hidden;
+  position: relative;
+  padding-bottom: ${(mockup.desktop.slide.height /
+    mockup.desktop.container.width) *
+    100}%;
+`
+
+const PrevNextSection = styled.div``
+
+const DescriptionSection = styled.div``
+
+const SlidesFlexBox = styled.div`
+  position: absolute;
+  top: 0;
+  left: 0;
+  display: flex;
+  flex-wrap: nowrap;
+  width: 100%;
+  height: 100%;
+  ${props => {
+    if (props.isSliding) {
+      return `transition: transform ${props.duration}ms ease-in-out;`
+    }
+  }}
+  transform: translateX(${props =>
+    getTranslateX(mockup.desktop, props.translateXUint)}px);
+`
+
+const SlideFlexItem = styled.div`
+  flex-shrink: 0;
+  flex-basis: ${getSlideWidth(mockup.desktop)}px;
+  padding-right: ${mockup.desktop.slide.paddingRight}px;
+`
+
+const SlideMask = styled.div`
+  position: absolute;
+  top: 0;
+  height: 100%;
+  opacity: 0.55;
+  background-image: linear-gradient(to bottom, #2440fb, #2440fb);
+`
+
+const LeftSlideMask = styled(SlideMask)`
+  left: 0;
+  width: ${getLeftMaskWidth(mockup.desktop)}px;
+`
+
+const RightSlideMask = styled(SlideMask)`
+  right: 0;
+  width: ${getRightMaskWidth(mockup.desktop)}px;
+`
+
+/**
+ * @typedef {Object} SlideMockup
+ * @property {number} width
+ * @property {number} height
+ * @property {number} paddingRight
+ */
+
+/**
+ * @typedef {Object} ContainerMockup
+ * @property {number} width
+ */
+
+/**
+ * @typedef {Object} OffsetMockup
+ * @property {number} left
+ */
+
+/**
+ * @typedef {Object} DeviceMockup
+ * @property {SlideMockup} slide
+ * @property {ContainerMockup} container
+ * @property {OffsetMockup} offset
+ */
+
+/**
+ * @param {DeviceMockup} deviceMockup
+ * @param {number} unit
+ * @return {number}
+ */
+function getTranslateX(deviceMockup, unit) {
+  const slideWidthWithPadding =
```

Comment:
If we take slide width as a constants, we can't use image with different width (ex: potrait image) in slide show, and may have to get screen width with js in modile device since its container will be responsive with screen size. But I've not figured out how to enchance this yet,

Reply:
My plan is to fix the width of slides, no matter what images would be(portrait or landscape).

According to gina's suggestion, she wants portrait image to be `contain`ed in the `SlidesSection`.

My current plan represents in the pseudo code:
```
if image.height > image.width
  image.object-fit = 'contain'
else 
  image.object-fit = 'cover'
```